azkaban-common/src/main/java/azkaban/imagemgmt/models/ImageVersion.java

```diff
@@ -165,13 +168,31 @@ public static State fromStateValue(final String stateValue) {
             .filter(state -> (state.equals(State.NEW) || state.equals(State.ACTIVE)))
             .collect(Collectors.toSet());
 
+    /**
+     * Create set of new, active and test state
+     */
+    private static final Set<State> newActiveAndTestState =
+        Arrays.stream(State.values())
+            .filter(state -> (state.equals(State.NEW) || state.equals(State.ACTIVE)
```

Comment:
Instead of streaming through whole enum set, how about
Set<State> set = new HashSet<>();
    Collections.addAll(set, State.NEW, State.ACTIVE, State.TEST);
    ?

Reply:
The idea behind this code is to precompute the set of states and keep them in the static variable. Whenever we need to use the filter simply return the state filter. As this list is fixed and part of the enum it is also behaving like how enums are defined and initialized only once. As the states are only a selected few, there won't be any issue if we stream and filter the required states (done only once). On the other hand with the suggested code we have to compute every time whenever the the filter is required (this filter is frequently used during flow dispatch).